File: LeetCodeRecord/MethodFirst/283-MoveZeros.java
package MethodFirst;

class MoveZeros {
    public void moveZeros(int[] nums) {
        int i = 0;
        int j = 0;

        while (j <= nums.length - 1) {
            if (nums[j] != 0) {
                int temp = nums[i];
                nums[i] =  nums[j];
                nums[j] = temp;
                i++;
            }

            j++;
        }
    }    

}

File: LeetCodeRecord/ACFirst-Golang/283-MoveZeros.go
package main

func moveZeroes(nums[] int) {
	numsLen := len(nums)
	
	if numsLen == 0 {
		return
	}

	quickMove(nums, 0, len(nums) - 1)
}


func quickMove(nums []int, left, right int) {
	if left >= right {
		return
	}

	i := left
	j := left

	for j <= right {
		if nums[j] != 0 {
			temp := nums[i]
			nums[i] = nums[j]
			nums[j] = temp
			i++
		}
		j++
	}
}